### PR TITLE
refactor(api): hw controller dir for clarity and avoiding circ deps

### PR DESCRIPTION
**Pull request by b-cooper**
## overview

As a part of debugging the pr branch for #4870, it became clear that there were a couple of rearrangements that would up the maintainability of the hardware controller and minimize the possibility of introducing circular dependencies within that module as it is updated in the following changes.
 
## changelog

- pull `API(HardwareAPILike)` out of `hardware_control.__init__.py` and put it in it's own `api.py` file
- move hw ctl constants into a `constants.py` file
- centralize hw ctl utils into a `utils.py` file
- centralize ht ctl types into a `types.py` (except the synthesis type `InstrumentByMount` in api.py to avoid `api.py` -> `pipettes.py` -> `types.py`  -> `pipettes.py`) circular dependency
- export only externally required objects from hw ctl's `__init__.py`

- similarly reorganize `hardware_control/modules/`
- fix a couple typos
- split up TC execute profile into two functions 

## review requests

- [ ] Does this refactor make sense?
- [ ] Does everything still behave as expected.

### CODE REVIEW

File: api/src/opentrons/hardware_control/util.py [hunk 1]
```diff
@@ -16,3 +19,17 @@ def use_or_initialize_loop(loop: Optional[asyncio.AbstractEventLoop]
     checked_loop = loop or asyncio.get_event_loop()
     checked_loop.set_exception_handler(_handle_loop_exception)
     return checked_loop
+
+
+def log_call(func):
+    if inspect.iscoroutinefunction(func):
```
Comment (amitlissack) [inline api/src/opentrons/hardware_control/util.py]: is this a decorator that just logs function calls on a class instance that has a `_log` member? 

Is it used anywhere?

Comment (b-cooper) [reply api/src/opentrons/hardware_control/util.py]: Yes, that's what it is. It used to be collocated with the `API` class in the `__init__` file, so I just moved it to the `utils` file for organization's sake.  Within the API class this decorator is used ~25 times currently.

Comment (sfoster1) [reply api/src/opentrons/hardware_control/util.py]: i had originally made it when building the skeleton of the hardware controller and i would be ok with deleting it honestly. it does not serve much value

File: api/src/opentrons/protocol_api/util.py [hunk 2]
```diff
@@ -251,7 +251,7 @@ def clamp_value(
         return max_value
     if input_value < min_value:
         MODULE_LOG.info(
-            f'{log_tag} calmped input {input_value} to {min_value}')
+            f'{log_tag} clamped input {input_value} to {min_value}')
```
Comment (sfoster1) [inline api/src/opentrons/protocol_api/util.py]: 🗜

**Review by sfoster1 — APPROVED** [review]
Style and idea LGTM, happy with it

File: api/src/opentrons/hardware_control/util.py (hunk 1 above)
Comment (amitlissack) [reply api/src/opentrons/hardware_control/util.py]: If it doesn't serve much value then let's delete away.

Comment (b-cooper) [reply api/src/opentrons/hardware_control/util.py]: I'll rip it out in the following PR then.

**Pull request merged**